python-manual/modules/ROOT/pages/migration.adoc

```diff
@@ -0,0 +1,333 @@
+
+Any of the 5.x versions of the driver is compatible both with Neo4j server 4.4 and 5.x, so you can upgrade the driver before you upgrade the server.
+
+[TIP]
+When upgrading the Neo4j server to a newer version, the Cypher queries in your application may also need updating. +
+See link:https://neo4j.com/docs/cypher-manual/current/deprecations-additions-removals-compatibility/[Cypher -> Deprecations, additions, and compatibility].
+
+
+== Recommended steps
+
+Using the migration assistent, the recommended upgrade steps are:
+
+1. Run the migration assistent with its default configuration and address all the issues that it surfaces. The `path` argument supports globbing and can be repeated as many times as needed. +
++
+[source, bash]
+----
+python main.py -l python <path...>
+----
++
+2. Re-run the assistent with the regex parser and check if any more legit issues are raised. +
++
+[source, bash]
+----
+python main.py -l python --regex-parser <path...>
+----
++
+The regex parser is able to parse implicit function calls and other hard-to-parse expressions, but is likely surface more false positive (for more information, see link:https://github.com/stefano-ottolenghi/neo4j-drivers-migration-assistent?tab=readme-ov-file#accuracy[Neo4j Drivers Migration Assistent -> Accuracy]).
+You can add any spurious entry to the ignored list, so that the assistent won't bring them up in future runs.
+3. Although the assistent can detect most of the locations where action is needed, a few changelog entries can't be surfaced in this form, so you should still revise the xref:breaking-changes[breaking changes] section and audit your codebase for the changes marked with {cross-mark}.
+4. You are ready to upgrade the driver package to the new version. +
++
+[source, bash]
+----
+pip install -U neo4j
+----
+
+
+== New features
+
+[cols="1a,1a"]
+|===
+
+|
+.Support for Python 3.12
+[%collapsible]
+====
+The driver is compatible with any Python version starting from 3.6 up to 3.12.
+====
+|
+.`asyncio` support
+[%collapsible]
+====
+For using the driver's features in parallel via the link:https://docs.python.org/3/library/asyncio.html[asyncio] package.
+
+See xref:concurrency.adoc[].
+====
+
+|
+.`Driver.execute_query()` to run transactions with less knowledge of driver's internals
+[%collapsible]
+====
+The new function is a wrapper for `Session.execute_read/write()`, but it abstracts away the result processing part and returns a list of records to the caller directly.
+
+See xref:query-simple.adoc[].
+====
+|
+.Rust extension for performance
+[%collapsible]
+====
+The link:https://github.com/neo4j-drivers/neo4j-python-driver-rust-ext[Rust extension to the Python driver] is an alternative driver package that yields a 3x to 10x speedup compared to the regular driver.
+You can install it with `pip install neo4j-rust-ext`, either alongside the `neo4j` package or as a replacement to it.
```

Comment:
```suggestion
You can install it with `pip install neo4j-rust-ext`,  as a replacement to the `neo4j` package.
```
Written like that in the README for a purpose :) currently works to have both, might stop or become weird in the future.

Reply:
This falls in the conversation we had a few weeks ago on what the plans for the two packages are. 
If the two packages are going to stay separate long-term, and the rust extension is going to be supported long-term, then your change is good; but if in 1-2 years we decide to make the rust extension the default, then people may end up with a dangling dependency. 

Has the roadmap got clearer?

Reply:
Looks like we're doubling down on making it work the way you've documented it here.

See also https://github.com/neo4j/neo4j-python-driver-rust-ext/issues/20 and https://github.com/neo4j/neo4j-python-driver-rust-ext/pull/22/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R23-R24